### PR TITLE
Feature/add-v4-standard-models - Add some standard models for future use.

### PR DESCRIPTION
**Pull request by deeleeramone**
There is a need to preload some standard provider models for things which do not exist yet, but we can reasonably assume they will in the near future.  Certainly not a complete - or even thorough - list so far, but something to get the conversation and action started.

### CODE REVIEW

**Review by the-praxs — COMMENTED** [review]
I think we could add this as a Hacktoberfest issue so that contributors can make use of it? Wdyt?

**Review by the-praxs — APPROVED** [review]
LGTM

**Comment by the-praxs** [comment]
> > I think we could add this as a Hacktoberfest issue so that contributors can make use of it? Wdyt?
> 
> The purpose would be to have standard models which already exist in the main library, which would help facilitate Hacktoberfest contributions and to avoid them being created multiple times.

Do you want to merge it then?

File: openbb_platform/platform/provider/openbb_provider/standard_models/index_info.py
Comment (piiq) [inline openbb_platform/platform/provider/openbb_provider/standard_models/index_info.py:38]: Can we either explain what nb_ means in the description or just use constituents?

File: openbb_platform/platform/provider/openbb_provider/standard_models/industry_pe.py
Comment (piiq) [inline openbb_platform/platform/provider/openbb_provider/standard_models/industry_pe.py:12]: Can we have `IndustryPEData` in this file?

File: openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py
Comment (piiq) [inline openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py:32]: Can we use full names instead of abbreviations here?

File: openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py
Comment (piiq) [inline openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py:33]: And likely "three_year" is not "One-year return"

File: openbb_platform/platform/provider/openbb_provider/standard_models/sector_pe.py
Comment (piiq) [inline openbb_platform/platform/provider/openbb_provider/standard_models/sector_pe.py:12]: Same as with IndexPe. Can we do `SectorPE` here?

**Review by piiq — CHANGES_REQUESTED** [review]
Thanks this is a very good PR
I've cherry picked the commit that should make codespell pass but there are some more linting errors that we better fix before merging

pylint complains about the absense of a classmethod decorator and mypy complains about `openbb_platform/platform/provider/openbb_provider/standard_models/index_sectors.py:16: error: No overload variant of "field_validator" matches argument types "str", "bool", "bool", "bool"  [call-overload]` etc

I've also added minor comments about the naming

File: openbb_platform/platform/provider/openbb_provider/standard_models/industry_pe.py
Comment (the-praxs) [reply openbb_platform/platform/provider/openbb_provider/standard_models/industry_pe.py:12]: Second this. It should be PE.

File: openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py
Comment (the-praxs) [reply openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py:32]: Second this. Fields need to be more descriptive.

File: openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py
Comment (the-praxs) [reply openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py:33]: Simple fix.

File: openbb_platform/platform/provider/openbb_provider/standard_models/index_info.py
Comment (deeleeramone) [reply openbb_platform/platform/provider/openbb_provider/standard_models/index_info.py:38]: We could make this, `num_constituents` if that is more clear?

File: openbb_platform/platform/provider/openbb_provider/standard_models/industry_pe.py
Comment (deeleeramone) [reply openbb_platform/platform/provider/openbb_provider/standard_models/industry_pe.py:12]: Perhaps this is a stylistic choice we should define somewhere in the contributing guidelines.  If we were to make the class name a representation of the full name, `IndustryPeRatio` is more readable than `IndustryPERATIO`

File: openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py
Comment (deeleeramone) [reply openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py:32]: YTD, TTM, etc. are all very common field names. Would expanding the description suffice?

File: openbb_platform/platform/provider/openbb_provider/standard_models/sector_pe.py
Comment (deeleeramone) [reply openbb_platform/platform/provider/openbb_provider/standard_models/sector_pe.py:12]: For some additional context, `FMPETFNAVQueryParams`, would be difficult to read vs. `FmpEtfNavQueryParams`

File: openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py
Comment (the-praxs) [reply openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py:32]: brevity vs verbosity but whichever you find the optimum choice gentlemen

File: openbb_platform/platform/provider/openbb_provider/standard_models/index_info.py
Comment (piiq) [reply openbb_platform/platform/provider/openbb_provider/standard_models/index_info.py:38]: yes this is way better

File: openbb_platform/platform/provider/openbb_provider/standard_models/industry_pe.py
Comment (piiq) [reply openbb_platform/platform/provider/openbb_provider/standard_models/industry_pe.py:12]: Camel case is when the first letter of each word is capitalized

ChatGPT agreed with me

image

File: openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py
Comment (piiq) [reply openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py:32]: Yes, I my ask was about to the descriptions

File: openbb_platform/platform/provider/openbb_provider/standard_models/sector_pe.py
Comment (piiq) [reply openbb_platform/platform/provider/openbb_provider/standard_models/sector_pe.py:12]: Yes it would.
Special cases aren't special enough to break the rules, although practicality beats purity

File: openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py
Comment (deeleeramone) [reply openbb_platform/platform/provider/openbb_provider/standard_models/recent_performance.py:33]: Added some more common look back periods.

**Comment by deeleeramone** [comment]
> there are some more linting errors that we better fix before merging

I do not get this linting error locally:

```
openbb_platform/platform/core/tests/api/test_auth/test_user_auth.py:2:1: I001 [*] Import block is un-sorted or un-formatted
```

**Comment by the-praxs** [comment]
> > there are some more linting errors that we better fix before merging
> 
> I do not get this linting error locally:
> 
> ```
> openbb_platform/platform/core/tests/api/test_auth/test_user_auth.py:2:1: I001 [*] Import block is un-sorted or un-formatted
> ```

Maybe it's the different version used in the CI and your local one?

**Comment by the-praxs** [comment]
@deeleeramone I think we should have `ETF` instead of `Etf` as its a very specific word. Wdyt?

**Comment by deeleeramone** [comment]
> @deeleeramone I think we should have `ETF` instead of `Etf` as its a very specific word. Wdyt?

I disagree.  Everything will then have TOOMANYCaps and be difficult to read.


FMPETF, TMXETF, etc.

**Review by piiq — APPROVED** [review]
I've resolved the linter errors. Feel free to merge it whenever you're ready

**Comment by deeleeramone** [comment]
> I've resolved the linter errors. Feel free to merge it whenever you're ready

Had to fix a conflict with upstream and the linting error is back!

**Comment by piiq** [comment]
For some reason ruff is not picking up the settings from pyproject.toml to exclude the package folder. Still trying to make it work

Additionally I've ran into linting problems that touch the OECD files that use syntax that's not supported in all python versions. My local linter in a python 3.9 env complains about them a lot. This is not catched here in the CI

@jmaslek you seen this, right?
image

**Comment by jmaslek** [comment]
> For some reason ruff is not picking up the settings from pyproject.toml to exclude the package folder. Still trying to make it work
> 
> Additionally I've ran into linting problems that touch the OECD files that use syntax that's not supported in all python versions. My local linter in a python 3.9 env complains about them a lot. This is not catched here in the CI
> 
> @jmaslek you seen this, right? image

Well you showed me elsewhere :P

This is what I get for using 3.11.  Think igor is patching it.  Worst case, we just copy paste the long list.  Was trying to keep that outside

**Comment by deeleeramone** [comment]
@piiq, this is getting beyond scope of PR, should we address linting errors upstream instead of this PR?

**Comment by the-praxs** [comment]
> @piiq, this is getting beyond scope of PR, should we address linting errors upstream instead of this PR?

I agree. The OECD linting issue should be a separate PR. This can be merged if all looks good for what this PR does exactly.

**Review by IgorWounds — APPROVED** [review]

**Pull request merged**
